<a href="https://colab.research.google.com/github/cvitorc/Exemplos-Estatistica-Mestrado/blob/main/Teste_ANOVA_com_Tukey_arquivo_xls_Exemplo02.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
!pip install statsmodels

In [ ]:
import pandas as pd
import statsmodels.api as sm
from statsmodels.formula.api import ols
from statsmodels.stats.multicomp import pairwise_tukeyhsd

# Leia o arquivo Excel (substitua 'seuarquivo.xlsx' pelo caminho do seu arquivo)
dados = pd.read_excel('exemplo-ANOVA2.xlsx')

# Realize a ANOVA
modelo = ols('Valor ~ Grupo', data=dados).fit()
tabela_anova = sm.stats.anova_lm(modelo, typ=2)

# Exiba os resultados da ANOVA
print("Resultados da ANOVA:")
print(tabela_anova)

# Extraia o valor de p da ANOVA
valor_p_anova = tabela_anova['PR(>F)'][0]

nivel_significancia = 0.05

# Exiba o valor de p da ANOVA
print("\nValor de p da ANOVA:", valor_p_anova)

# Verifique se o valor de P e maior que o nível de significancia
if valor_p_anova > nivel_significancia:
    print(f"\nA ANOVA não é significativa (p-valor: {valor_p_anova}), não rejeitando a hipótese nula.")
else:
    print(f"\nA ANOVA é significativa (p-valor: {valor_p_anova}), rejeitando a hipótese nula.")

    # Realize o teste de Tukey
    resultado_tukey = pairwise_tukeyhsd(endog=dados['Valor'], groups=dados['Grupo'])

    # Exiba os resultados do teste de Tukey
    print("\nResultados do Teste de Tukey:")
    print(resultado_tukey.summary())



Resultados da ANOVA:
             sum_sq    df        F    PR(>F)
Grupo     14.533333   2.0  4.27451  0.039659
Residual  20.400000  12.0      NaN       NaN

Valor de p da ANOVA: 0.03965891577699042

A ANOVA é significativa (p-valor: 0.03965891577699042), rejeitando a hipótese nula.

Resultados do Teste de Tukey:
Multiple Comparison of Means - Tukey HSD, FWER=0.05
group1 group2 meandiff p-adj  lower upper reject
------------------------------------------------
  café   suco      1.0 0.4685  -1.2   3.2  False
  café   água      2.4 0.0325   0.2   4.6   True
  suco   água      1.4 0.2459  -0.8   3.6  False
------------------------------------------------


Resposta:

Os resultados indicam que, com um nível de significância de 0.05 (FWER=0.05), a análise de variância (ANOVA) encontrou diferenças significativas entre os grupos, e o teste de Tukey revelou quais pares de grupos são significativamente diferentes. Vamos interpretar esses resultados:

Resultados da ANOVA:

A estatística F é 4.27451, e o valor p associado a ela é 0.039659. Isso indica que o valor p (PR(>F)) é menor que o nível de significância de 0.05. Portanto, há evidências para rejeitar a hipótese nula, sugerindo que existem diferenças significativas entre os grupos.
Resultados do Teste de Tukey:

O teste de Tukey compara pares de grupos para determinar quais deles são estatisticamente diferentes. Os resultados do teste de Tukey são exibidos em uma tabela:

Entre "café" e "suco", o valor p ajustado (p-adj) é 0.4685, que é maior que 0.05, indicando que não há diferença significativa entre esses grupos. A coluna "reject" está marcada como "False", confirmando que esses grupos não são significativamente diferentes.

Entre "café" e "água", o valor p ajustado é 0.0325, que é menor que 0.05, indicando uma diferença significativa. A coluna "reject" está marcada como "True", confirmando a diferença significativa entre esses grupos.

Entre "suco" e "água", o valor p ajustado é 0.2459, que é maior que 0.05, indicando que não há diferença significativa. A coluna "reject" está marcada como "False", indicando que esses grupos não são significativamente diferentes.

Portanto, com base nesses resultados, podemos concluir que existe uma diferença significativa entre os grupos "café" e "água", mas não entre os outros pares de grupos.
